### DISCUSSION TITLE
Some Blog Archive feature questions

### DISCUSSION THREAD

**Discussion by Andre601** [squidfunk/mkdocs-material#5820] [Q&A] [answered]
I've read through the [Section about setting up a blog archive](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#archive) but I still have a few questions I would like to receive answers for.

The first one is: How does the blog plugin determine a blog post to be "archived". Will it be every post that is in a previous year than the current? Any that is X days, weeks, months, ... old?
It would help me understand this, as I think an archive *could* be useful for me to have for my blog...

Next question: Does the archiving of a post affect its effective URL path?
I ask this because I fear archiving existing blog posts could mess with their path, making something like `/blog/2023/01-01-something-blabla/` no longer work if the post gets archived.
Does the archiving only make the index pages (And maybe add some indicator to the post) but keep the URL the same, similar to tags and categories?

And final question: Would setting the `archive_url_format` to `{date}` cause issues if the posts are under a `yyyy` path?
I have my posts under a year folder, so `2020/...`, `2021/...`, etc. for better structuring. I now wonder if I could *exploit* using `{date}` as `archive_url_format` option to generate Archive indexes for these year sub-pages, because as of Right now are those leading to a 404...

(Alternatively the above could be considered some kind of feature request to auto-generate sub-pages for posts within subfolders, so that accessing `2023/` would list all posts within the `2023` folder...)

**Comment by squidfunk** [comment, marked as answer]
You don't explicitly archive blog posts. The blog archive works the same as categories, except for that it uses the `date` value of a post with the format you specified for classification. Every archive page includes all pages that fall into the date range (yearly, or monthly, depending on how you configured [`archive_url_date_format`](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#+blog.archive_url_date_format)), except for when you mark them as drafts. This means that the 2023 archive when using the yearly date format will always contain all posts of 2023.

As our documentation [clearly states](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#writing-your-first-post), file system structure has absolutely no effect on generated URLs:

> After you've successfully set up the [built-in blog plugin](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#built-in-blog-plugin), it's time to write your first post. The plugin doesn't assume any specific directory structure, so you're completely free in how you organize your posts, as long as they are all located inside the posts directory:

You can move posts around as you wish, group them intelligently so that you can make use of the [built-in `meta` plugin](https://squidfunk.github.io/mkdocs-material/reference/#built-in-meta-plugin) to auto-tag or -categorize posts in certain folders. Post URLs are always computed from [`post_url_format`](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#+blog.post_url_format).

**Reply by Andre601** [reply]
Okay, thanks.

What about my question regarding the use of just `{date}` as the `archive_url_format` option?
As mentioned do I have my posts within a folder named after the year within the posts directory (i.e. file `01-01-some-post.md` is located at `docs/posts/2023/01-01-some-post.md`), so the posts show up under `/<year>/<file-name>` in the final page (I use file name as the slug instead of post title).

And I was wondering if setting the `archive_url_format` to `{date}` with `yyyy` being used as date would allow me to have some index pages for just `/<year>` as those pages right now lead to a 404 page...
I just fear there could be conflicts, although I doubt that this would be the case.

**Reply by squidfunk** [reply]
Unless you don't have a `post_url_format` that resolves to `/2023/`, I see no problem. If in doubt, you can just, well, try it out?

**Reply by Andre601** [reply]
`post_url_format` is set to `{date}/{file}` so it should be fine I assume
